python/lsst/ts/hvac/xml/hvac_mqtt_to_SAL_XML.py

```diff
@@ -19,153 +19,495 @@
-                topic_found = True
-                break
-
-            item = re.sub(hvac_topic.value, "", topic)
-            if hvac_topic.name not in telemetry_topics:
-                telemetry_topics[hvac_topic.name] = {}
 
-            item_found = False
-            for telemetry_item in TelemetryItem:
-                if telemetry_item.value == item:
-                    idl_type = "boolean" if re.match(r"BINARY", row[5]) else "float"
-                    telemetry_topics[hvac_topic.name][telemetry_item.name] = idl_type
-                    item_found = True
-                    break
-            if not item_found:
-                raise ValueError(f"TelemetryItem '{item}' for {topic} not found.")
+def extract_topic_and_item(topic_and_item):
+    """Extract the generic topic, representing a HVAC subsystem, and item,
+     representing a published value of the subsystem, from a string.
 
-            topic_found = True
-            break
-    if not topic_found:
-        raise ValueError(f"Topic {topic} not found.")
+    This method searches for the last occurrence of that forward slash and will
+    return the part before the slash as topic and after the slash as item.
 
+    Parameters
+    ----------
+    topic_and_item: `str`
+        The string to extract the generic topic and item from.
 
-def collect_command_topics_and_items(row):
-    global command_topics
-    global commands
+    Returns
+    -------
+    topic: `str`
+        Defined as the part before the last forward slash.
+    item: `str`:
+         Defined as the part after the last forward slash.
+
+    Raises
+    ------
+    ValueError
+        In case no forward slash is found.
+    """
+    # This throws a ValueError in case no forward slash is found.
+    topic_end = topic_and_item.rindex("/")
+    # omit the forward slash
+    item_start = topic_end + 1
+    topic = topic_and_item[:topic_end]
+    item = topic_and_item[item_start:]
+    return topic, item
+
+
+def get_topics():
+    """Convenience method to collect all generic topics, representing the HVAC
+    subsystems.
+
+    Returns
+    -------
+    topics: `set`
+        A set of all generic topics.
+
+    """
+    global hvac_topics
+    if not bool(hvac_topics):
+        collect_topics_and_items()
+    topics = set()
+    for topic_and_item in hvac_topics:
+        topic_type = hvac_topics[topic_and_item]["topic_type"]
+        if topic_type == "WRITE" and topic_and_item.endswith("COMANDO_ENCENDIDO_LSST"):
+            topic, item = extract_topic_and_item(topic_and_item)
+            topics.add(topic)
+    for topic in TOPICS_ALWAYS_ENABLED:
+        topics.add(topic)
+
+    return topics
+
+
+def get_telemetry_mqtt_topics_and_items():
+    """Convenience method to collect all MQTT topics and their items that
+    publish telemetry.
+
+    Returns
+    -------
+    telemetry_mqtt_topcis: `dict`
+        A dictionary with the MQTT topics as keys and their items as value.
```

Comment:
I agree that this code is not very clear mainly due to the overloaded use of both topic and item in MQTT and SAL. I'll consider rewriting this code at some point but for now I'll leave it as is since modifying it too much will bring on more work and I already spent so much more time on this then intended.